tdp/core/collections.py

```diff
@@ -137,87 +137,115 @@ def operations(self) -> dict[str, Operation]:
             operations.update(self._other_operations)
         return operations
 
-    def _init_operations(self):
-        self._dag_operations = {}
-        self._other_operations = {}
-
-        # Init DAG Operations
-        for collection_name, collection in self._collections.items():
-            for dag_yaml_file in collection.dag_yamls:
-                operations_list = None
-                operations_list = read_tdp_lib_dag_file(dag_yaml_file)
-
-                for operation in operations_list:
-                    if operation.name in self._dag_operations:
-                        # Merge the operation with the existing one
+    def _init_operations(
+        self, collections: Mapping[str, Collection]
+    ) -> tuple[dict[str, Operation], dict[str, Operation]]:
+        dag_operations: dict[str, Operation] = {}
+        other_operations: dict[str, Operation] = {}
+
+        for collection in collections.values():
+            # Load DAG operations from the dag files
+            for dag_file in collection.dag_yamls:
+                for read_operation in read_tdp_lib_dag_file(dag_file):
+                    existing_operation = dag_operations.get(read_operation.name)
+
+                    # The read_operation is a playbook operation defined in the current
+                    # collection
+                    if playbook_operation := collection.playbooks.get(
+                        read_operation.name
+                    ):
+                        # Create an Operation instance based on the playbook
+                        # TODO: would be nice to dissociate the Operation class from the playbook and store the playbook in the Operation
+                        dag_operation_to_register = Operation(
+                            name=read_operation.name,
+                            collection_name=playbook_operation.collection_name,
+                            host_names=playbook_operation.host_names,
+                            depends_on=read_operation.depends_on,
```

Comment:
`depends_on` is an array so the ref is given to the new Operation which is wrong because when using `.extend` it will update the `depends_on` for the `read_operation` too. You must use `deepcopy` for example.

Maybe it will be better to have a `clone` method for `Operation` which deep clone an operation.